doc/rst/usingchapel/QUICKSTART.rst

```diff
@@ -117,40 +116,42 @@ you will need to rebuild Chapel from source in a different configuration.
       # re-build Chapel
       make
 
-      # make check is available but optional
-      make check
-
       # compile a sample program
-      chpl -o hello examples/hello.chpl
+      chpl -examples/hello3-datapar.chpl
 
       # run the sample program
-      ./hello
+      ./hello3-datapar
 
-   Note that the environment settings from ``util/setchplenv.bash`` will not persist beyond this terminal session.
-   You can choose to source ``setchplenv.bash`` whenever you want to use relevant commands.
-   If you want these environment settings to persist for future terminal sessions,
-   copy the commands from the Recommended Settings in :ref:`chplenv.rst <readme-chplenv.recommended_settings>` into your ``~/.bashrc`` file.
-   You can also store Chapel configuration settings in a :ref:`chplconfig <readme-chplenv.chplconfig>` file instead of your ``~/.bashrc``.
+   If you run into any portability issues, please see
+   :ref:`readme-bugs`:
 
-   See :ref:`chplenv.rst <readme-chplenv>` for a complete description of
-   Chapel's configuration variables, what they mean, and how they
-   can be set.
+
+Using Chapel in a Different Shell / Session
+-------------------------------------------
+
+Note that in both the Quickstart and preferred modes above, any
+environment settings made by ``setchplenv.bash`` will not persist
+beyond your current shell / terminal session.  To use Chapel from a
+different shell or terminal, you will need to either re-``source`` the
+``setchplenv.bash`` script that you used when building Chapel or set
+up your environment to support additional shells automatically.  See
+:ref:`using-chapel-in-other-sessions` for details.
```

Comment:
Is another layer of indirection needed here?  Can't they just say `make install`?

Reply:
I think you also have to run `./configure` before making anything in order for `make install` to work, which requires specifying one of the install path forms if you don't want the default of `/usr/local` (which I'd say isn't an appropriate default for a QUICKSTART user who's likely to get a permissions error and be confused by it).  And I don't really want the QUICKSTART to have to teach the user about all of that—at least this week and until/unless we have more / better testing of it and confidence in it.

Note that `./configure` and `make install` can be run after all of the instructions above with very little time spent or downsides (specifically, it won't generally re-do the whole `make` from scratch, so I don't think much is lost by mentioning it late / doing it after the fact).

Reply:
As it stands now, I worry that it will lead a user to think that installing is not possible. (Because it implies that the *only* way to have it work in different terminals is by sourcing something possibly from bashrc / changing environment variables).

In particular, I think part of the problem is that I interpret this section

```
To use Chapel from a
different shell or terminal, you will need to either re-``source`` the
``setchplenv.bash`` script that you used when building Chapel or set
up your environment to support additional shells automatically.  See
:ref:`using-chapel-in-other-sessions` for details.
```

as not including `make install` as a possibility. Perhaps you were imagining that `make install` is a way to "set up your environment"?

Instead, I think we should fix it by relying more on the linked text - replacing the above quoted section with this:

```
One easy way to use Chapel from a
different shell or terminal is to re-``source`` the
``setchplenv.bash`` script that you used when building Chapel.
However since that can be annoying to do, there are several
other strategies available to you. Please see :ref:`using-chapel-in-other-sessions`
to learn about them.
```